File: broker/pub/publisher.js
const Fast = require("fastify");
const zmq = require("zeromq");

const app = new Fast();
const sock = new zmq.Publisher();
sock.connect("tcp://localhost:5554");

app.post("/", async (request, reply) => {
  const msg = JSON.stringify({ ...request.body });
  console.log("Received: %s", msg);

  await sock.send(["pubsub", msg]);
  return reply.send("Sent to the subscriber/worker.");
});

const main = async () => {
  try {
    await app.listen({ port: 3000 });
  } catch (err) {
    console.error(err);
    process.exit(1);
  }
};
main();

File: ps/pub/publisher.py
# -*- coding: utf-8 -*-

import zmq
import sys
from flask import (request, Flask)

port = "5555"
if len(sys.argv) > 1:
    port =  sys.argv[1]
    int(port)

app = Flask(__name__)
context = zmq.Context()
socket = context.socket(zmq.PUB)

@app.route('/', methods={"POST"})
def hello():
  if request.method == 'POST':
    content_type = request.headers.get('Content-Type')
    if (content_type == 'application/json'):
      message = request.json
      socket.send("pubsub %s" % (message))
    return 'Sent to the subscriber/worker.'

if __name__ == "__main__":
  socket.bind("tcp://127.0.0.1:%s" % port)
  app.run(debug=True, port=3000)
